### DISCUSSION TITLE
How to access attributes of results.show and results.print in YOLOV%?

### DISCUSSION THREAD

**Discussion by rabiyaabbasi** [ultralytics/yolov5#7771] [Q&A]
Hi, I am using the below code to load the trained custom Yolov5 model and perform detections. 

import cv2
import torch
from PIL import Image
model = torch.hub.load('ultralytics/yolov5', 'custom', path='yolov5/runs/train/exp4/weights/best.pt', force_reload=True) 
img = cv2.imread('example.jpeg')[:, :, ::-1]  # OpenCV image (BGR to RGB)
results = model(img, size=416)  

To display and save results I am using:
results.print()  
results.save() 
results.show()

My question is how can I save the results in different directories so that I can use them in my web-based application (for your reference I am using Streamlit). For instance, at the moment results (image) are saved in runs\detect\exp*. I want to change it. Secondly, every time 'exp' is incremented and a new directory is created to save results. I want only one directory where the image is saved and a new image overwrites the previous one. I have tried to modify it in common.py but its not working. Last but not the least how can I access the results.print() attributes, so that I can use it for further manipulation. For instance, at the moment it prints (when run in jupyter notebook cell):

image 1/1: 1024x768 2 Car
Speed: 17.8ms pre-process, 30.7ms inference, 2.1ms NMS per image at shape (1, 3, 416, 320)

How can I access the detected class and perform further manipulation?

**Comment by glenn-jocher** [comment]
@rabiyaabbasi the .save() method has a save_dir argument:
https://github.com/ultralytics/yolov5/blob/4870064629e76d9d387578c562a292cc680fa05f/models/common.py#L654-L657

**Reply by rabiyaabbasi** [reply]
Thank you so much for your response.  I am trying to create a new directory with the following few changes. After running results.save, its again making same directory and saving results in run/detect/exp. Why is that? With reference to below code, .save() method should run save_dir argument and make directory name as Folder/Detect.

 def save(self, labels=True, save_dir='Folder/Detect'): 
     save_dir = increment_path(save_dir, exist_ok=save_dir != 'Folder/Detect', mkdir=True)  # increment save_dir 
     self.display(save=True, labels=labels, save_dir=save_dir)  # save results

**Reply by glenn-jocher** [reply]
I have no idea what you are asking. .save() works correctly for custom save directories:

<img width="1517" alt="Screenshot 2022-05-11 at 23 29 47" src="https://user-images.githubusercontent.com/26833433/167951062-80975aa1-619f-4096-bdb9-1e8bc6609e32.png">

**Comment by rabiyaabbasi** [comment]
Hi 

Now I know where I am making mistake. It's working completely fine now. Thanks for your help.
